.circleci/config.yml

```diff
@@ -3,25 +3,34 @@ version: 2.1
 jobs:
```

Comment:
@MichaelDuffieldArcha I am missing the context of these changes therefore take my comment more like a proposal rather than request for change:

* Wouldn't be better to create another specific job related to these changes as they might impact the current solution?

Reply:
@lukas-aron-archa The changes here are primarily related to ensuring that the library runs with newer Python versions by:
- Updating the tests to ensure that the library is tested against newer python versions
- Replacing deprecated code
with the intention of ensuring that this code doesn't break as we upgrade our downstream services to use different Python versions.

The downstream services will still require a manual intervention to use the updated library code - and in the event of an unforseen catastrophic failure which isn't picked up by either the library tests here or the downstream tests, the downstream services could just be reverted to use the previous version of this library - so I don't see merging this updated code as a high-risk or irreversible operation.